paddle/fluid/operators/controlflow/fetch_op.cc

```diff
@@ -56,16 +56,20 @@ class FetchOp : public framework::OperatorBase {
     // FIXME(yuyang18): Should we assume the fetch operator always generate
     // CPU outputs?
     if (src_item.IsInitialized() && src_item.numel() > 0) {
+#ifdef PADDLE_WITH_MKLDNN
       // Conversion from MKL-DNN to Paddle
       if (src_item.layout() == framework::DataLayout::kMKLDNN) {
         framework::Tensor out;
         framework::innerTransDataLayoutFromMKLDNN(
-            src_item.layout(), framework::DataLayout::kNCHW, src_item, &out,
-            platform::CPUPlace());
+            src_item.layout(), paddle::platform::get_cur_paddle_data_layout(),
+            src_item, &out, platform::CPUPlace());
         TensorCopySync(out, platform::CPUPlace(), &dst_item);
       } else {
         TensorCopySync(src_item, platform::CPUPlace(), &dst_item);
       }
+#else
+      TensorCopySync(src_item, platform::CPUPlace(), &dst_item);
+#endif
```

Comment:
line72 is duplicated with line68

Reply:
I think it's ok. TensorCopySync is still needed in this place even without PADDLE_WITH_MKLDNN defined.

Reply:
@grygielski How about remove line66-68?

Reply:
That is correct , but when I tried to have this line only once ,then there were two #ifdef PADDLE_WITH_MKLDNN blocks and it was not very readable.

Reply:
Then it will result in completely different code. If you build with PADDLE_WITH_MKLDNN you get:
```
if (src_item.layout() == framework::DataLayout::kMKLDNN) {
    ...
} else {
    TensorCopySync(src_item, platform::CPUPlace(), &dst_item);
}
```
When you delete these lines you will get:
```
if (src_item.layout() == framework::DataLayout::kMKLDNN) {
    ...
}
```
If you want to delete also this `#else` it will result in:
```
if (src_item.layout() == framework::DataLayout::kMKLDNN) {
    ...
}
TensorCopySync(src_item, platform::CPUPlace(), &dst_item);
```

Reply:
Got it